src/Components/WebAssembly/Authentication.Msal/src/Interop/AuthenticationService.ts

```diff
@@ -125,7 +125,7 @@ class MsalAuthorizeService implements AuthorizeService {
     async getTokenCore(scopes?: string[]): Promise<AccessToken | undefined> {
         const account = this.getAccount();
         if (!account) {
-            return;
+            throw new Error('Failed to retrieve token, no account found.');
```

Comment:
This exception is caught on line 119 above in the caller function. By throwing/catching the exception we no longer return a `AccessTokenResultStatus.Success` but rather `AccessTokenResultStatus.RequiresRedirect`. This goes into:

https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Components/WebAssembly/WebAssembly.Authentication/src/Services/AccessTokenResult.cs#L50-L51

Resulting in:
https://github.com/dotnet/aspnetcore/blob/d7acbfb3c4d14f38eaaa213410e3c71752bdad9f/src/Components/WebAssembly/WebAssembly.Authentication/src/Services/AuthorizationMessageHandler.cs#L66

Verified this is working as expected locally.